src/test/java/core/basesyntax/service/FruitServiceTest.java

```diff
@@ -0,0 +1,110 @@
+package core.basesyntax.service;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.dao.FruitDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Transaction;
+import core.basesyntax.service.impl.FruitServiceImpl;
+import core.basesyntax.service.impl.OperationServiceImpl;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.impl.BalanceHandlerImpl;
+import core.basesyntax.strategy.impl.PurchaseHandlerImpl;
+import core.basesyntax.strategy.impl.ReturnHandlerImpl;
+import core.basesyntax.strategy.impl.SupplyHandlerImpl;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FruitServiceTest {
+    private static final FruitDao fruitDao = new FruitDaoImpl();
+    private static final List<Transaction> transactions = new ArrayList<>();
+    private static FruitService fruitService;
+
+    @BeforeClass
+    public static void beforeClass() {
+        Map<Transaction.Operation, OperationHandler> mapStrategy = new HashMap<>();
+        mapStrategy.put(Transaction.Operation.BALANCE, new BalanceHandlerImpl(fruitDao));
+        mapStrategy.put(Transaction.Operation.SUPPLY, new SupplyHandlerImpl(fruitDao));
+        mapStrategy.put(Transaction.Operation.PURCHASE, new PurchaseHandlerImpl(fruitDao));
+        mapStrategy.put(Transaction.Operation.RETURN, new ReturnHandlerImpl(fruitDao));
+        OperationService operationService = new OperationServiceImpl(mapStrategy);
+        fruitService = new FruitServiceImpl(operationService);
+    }
+
+    @Test
+    public void setBalance_ok() {
+        transactions.add(new Transaction(
+                Transaction.Operation.BALANCE, "apple", 23));
+        fruitService.process(transactions);
+        int expected = 23;
+        int actual = fruitDao.getQuantity("apple");
```

Comment:
don't use DAO here, you can access storage directly
when we test some unit (for example FruitService) we shouldn't depend on some other unit, this way different logic are not mixed up

Reply:
not only here, everywhere in your project